File: bt2.11trang34.cpp
#include <bits/stdc++.h>
using namespace std;
int fibo(int n){
	int f1=0,f2=1;
	while (f2<=n){
		int fn=f1+f2;
		f1=f2;
		f2=fn;
	}
	return f1;
}
int main(){
	int n;cin>>n;
	while (n>0){
		cout<<fibo(n)<<" ";
		n-=fibo(n);
	}
}

File: bt2.11trang34.py
def fibo(n):
    f1=0
    f2=1
    while (f2<=n):
        fn=f1+f2
        f1=f2
        f2=fn
    return f1
n=int(input())
while (n>0):
    print(fibo(n), end=' ')
    n=n-fibo(n)
